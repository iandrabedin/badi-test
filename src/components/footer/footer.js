import React from "react";
import "./footer.scss";

export const Footer = () => (
  <footer className="footer">
    <div className="container">
      Made with
      <a href="https://reactjs.org/" target="_blank" rel="noopener noreferrer">
        ReactJS
      </a>
      and
      <a
        href="http://www.recipepuppy.com/about/api/"
        target="_blank"
        rel="noopener noreferrer"
      >
        Recipe Puppy API
      </a>
    </div>
  </footer>
);
