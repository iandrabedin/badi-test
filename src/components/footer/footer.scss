@import "../../styles/variables.scss";

.footer {
  width: 100%;
  bottom: 0;
  text-align: center;
  height: 64px;
  display: flex;
  align-items: center;
  margin-top: 32px;

  a {
    padding: 0 4px;
  }
}
