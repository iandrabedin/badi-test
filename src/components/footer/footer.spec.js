import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Footer } from "./footer";

afterEach(cleanup);

describe("Footer", () => {
  it("should render as expect", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the correct links", () => {
    const { getByText } = render(<Footer />);
    const reactLink = getByText("ReactJS");
    const apiLink = getByText("Recipe Puppy API");
    expect(reactLink).toHaveAttribute("href", "https://reactjs.org/");
    expect(apiLink).toHaveAttribute(
      "href",
      "http://www.recipepuppy.com/about/api/"
    );
  });
});
