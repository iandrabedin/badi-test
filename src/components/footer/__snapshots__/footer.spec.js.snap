// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render as expect 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      Made with
      <a
        href="https://reactjs.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        ReactJS
      </a>
      and
      <a
        href="http://www.recipepuppy.com/about/api/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Recipe Puppy API
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
